misc/python/materialize/output_consistency/validation/result_comparator.py

```diff
@@ -0,0 +1,263 @@
+            )
+
+            # this needs will no longer be sensible when more than two evaluation strategies are used
+            self.remark_on_success_with_single_column(outcome1, validation_outcome)
+
+        if both_failed:
+            failure1 = cast(QueryFailure, outcome1)
+            self.validate_error_messages(
+                failure1,
+                cast(QueryFailure, outcome2),
+                validation_outcome,
+            )
+
+        if not both_successful:
+            any_failure = cast(
+                QueryFailure, outcome1 if not outcome1.successful else outcome2
+            )
+            validation_outcome.add_remark(
+                ValidationRemark(
+                    f"DB error in '{any_failure.strategy.name}' was: {any_failure.error_message}"
+                )
+            )
+            self.warn_on_failure_with_multiple_columns(any_failure, validation_outcome)
+
+    def validate_row_count(
+        self,
+        result1: QueryResult,
+        result2: QueryResult,
+        validation_outcome: ValidationOutcome,
+    ) -> None:
+        # It is ok if both results don't have any rows.
+
+        num_rows1 = len(result1.result_rows)
+        num_rows2 = len(result2.result_rows)
+
+        if num_rows1 != num_rows2:
+            validation_outcome.add_error(
+                ValidationError(
+                    ValidationErrorType.ROW_COUNT_MISMATCH,
+                    "Row count differs",
+                    value1=str(num_rows1),
+                    value2=str(num_rows2),
+                    strategy1=result1.strategy,
+                    strategy2=result2.strategy,
+                    sql1=result1.sql,
+                    sql2=result2.sql,
+                )
+            )
+
+    def validate_error_messages(
+        self,
+        failure1: QueryFailure,
+        failure2: QueryFailure,
+        validation_outcome: ValidationOutcome,
+    ) -> None:
+        norm_error_message_1 = self.normalize_error_message(failure1.error_message)
+        norm_error_message_2 = self.normalize_error_message(failure2.error_message)
+
+        if norm_error_message_1 != norm_error_message_2:
+            validation_outcome.add_error(
+                ValidationError(
+                    ValidationErrorType.ERROR_MISMATCH,
+                    "Error message differs",
+                    value1=norm_error_message_1,
+                    value2=norm_error_message_2,
+                    strategy1=failure1.strategy,
+                    strategy2=failure2.strategy,
+                    sql1=failure1.sql,
+                    sql2=failure2.sql,
+                )
+            )
+
+    def normalize_error_message(self, error_message: str) -> str:
+        # replace source prefix in column
+        normalized_message = re.sub(
+            'column "[^.]*\\.', 'column "<source>.', error_message
+        )
+        normalized_message = normalized_message.replace("Evaluation error: ", "")
```

Comment:
One of the two evaluation strategies uses this prefix, is this a relevant deviation that deserves a ticket?

Reply:
I think this is expected and is used to wrap errors which happen inside some evaluation context